src/pkg/services/m365/api/lists.go

```diff
@@ -253,8 +259,8 @@ func (c Lists) PostList(
 
 	if newList != nil &&
 		newList.GetList() != nil &&
-		ptr.Val(newList.GetList().GetTemplate()) == WebTemplateExtensionsListTemplateName {
-		return nil, clues.StackWC(ctx, ErrCannotCreateWebTemplateExtension)
+		NonRestorableListTemplates.HasKey(ptr.Val(newList.GetList().GetTemplate())) {
+		return nil, clues.StackWC(ctx, ErrCannotCreateNonRestorableListTemplate)
```

Comment:
I think the error could be something smaller like `SystemListTempalte` and we could have more information in the comments next to the const definition.

Reply:
documentLibrary will not fall under system generated. Lists with those template needs to handled differently. Putting documentLibrary under non-restorable is temporary change. Should we shorten it to 'ErrNonRestorableListTemplate'

Reply:
That sounds good, lets do `ErrNonRestorableListTemplate`.